app/src/main/java/ch/epfl/polychef/utils/Sort.java

```diff
@@ -31,38 +54,26 @@ public static void sortByIngredientSimilarity(List<Miniatures> list, String quer
     private static Comparator<Miniatures> similarityComparator = (o1, o2) -> {
         String s1 = o1.getName();
         String s2 = o2.getName();
-        if(Similarity.similarity(s1,actualQuery)>Similarity.similarity(s2,actualQuery)){
-            return -1;
-        }else if(Similarity.similarity(s1,actualQuery)==Similarity.similarity(s2,actualQuery)){
-            return 0;
-        }else {
-            return 1;
-        }
+
+        //noinspection UseCompareMethod (here we want to return exactly -1 or 1)
+        return Similarity.similarity(s1,actualQuery)>Similarity.similarity(s2,actualQuery) ? -1 :
+                Similarity.similarity(s1,actualQuery)==Similarity.similarity(s2,actualQuery) ? 0 : 1;
     };
 
     private static Comparator<Miniatures> similarityIngredientComparator = (o1, o2) -> {
-        List<Double> sim1 = Lists.transform(((Recipe)o1).getIngredients(), x-> Similarity.similarity(x.getName(),actualQuery));
-        List<Double> sim2 = Lists.transform(((Recipe)o2).getIngredients(), x-> Similarity.similarity(x.getName(),actualQuery));
+        List<Double> sim1 = Lists.transform(((Recipe)o1).getIngredients(), x-> Similarity.similarity(Objects.requireNonNull(x).getName(),actualQuery));
+        List<Double> sim2 = Lists.transform(((Recipe)o2).getIngredients(), x-> Similarity.similarity(Objects.requireNonNull(x).getName(),actualQuery));
         Double max1 = Collections.max(sim1);
         Double max2 = Collections.max(sim2);
-        if(max1>max2){
-            return -1;
-        }else if(max1==max2){
-            return 0;
-        }else {
-            return 1;
-        }
+
+        return max1>max2 ? -1 : max1.equals(max2) ? 0 : 1;
```

Comment:
Could replace with `Integer.compare(max2, max1)` ( or maybe the other way around?)

Reply:
I also asked myself the question but in the end in didn't do it since compare() will weaken the return assumption (returns positive or negative numbers, not exactly 1 or -1). 
Since I did not know if the exact values +1 / -1 were checked somewhere else I just skipped on it. But if someone checks we can adapt the code to the compare method.

Reply:
OK, we will have to ask Thomas then, since he is the one who did this originaly

Reply:
@wilderino ?

Reply:
I think compare will work ^^

Reply:
We just have do to it in the good way

Reply:
ok I changed it then